src/main/java/com/nutomic/syncthingandroid/preferences/WifiSsidPreference.java

```diff
@@ -115,7 +117,10 @@ private void filterRemovedNetworks(Set<String> selected, CharSequence[] all) {
                 Arrays.sort(result, new Comparator<WifiConfiguration>() {
                     @Override
                     public int compare(WifiConfiguration lhs, WifiConfiguration rhs) {
-                        return lhs.SSID.compareToIgnoreCase(rhs.SSID);
+                        // See #620: There may be null-SSIDs
+                        String l = lhs != null && lhs.SSID != null ? lhs.SSID : "";
+                        String r = rhs != null && rhs.SSID != null ? rhs.SSID : "";
+                        return l.compareToIgnoreCase(r);
```

Comment:
Is there any reason to check if the config is null? The issue (and your other fix) is only about the SSID being null, so I would only handle that. 

It's pretty annoying that Java doesn't have a standard method for this, but you could use [this code](http://stackoverflow.com/a/11271611) to shorten it (which also looks rather complicated).